src/java.desktop/share/classes/sun/font/SunFontManager.java

```diff
@@ -1445,8 +1446,10 @@ private void resolveFontFiles(HashSet<String> unmappedFiles,
     }
 
     Font2D findFontFromPlatformMap(String lcName, int style) {
+        HashMap<String, FamilyDescription> platformFontMap = SunFontManager.platformFontMap;
         if (platformFontMap == null) {
             platformFontMap = populateHardcodedFileNameMap();
+            SunFontManager.platformFontMap = platformFontMap;
```

Comment:
I am not sure it is enough for "thread-safe initialization". The "platformFontMap" might not be null, but its content can be broken/incomplete at this point.

Reply:
https://github.com/openjdk/jdk/pull/2691#issuecomment-784367127

In short, previously platformFontMap references were field set/gets and may be inconsistent due to concurrency. It is now moved to a local variable; also the map modifications have been moved to fields to avoid concurrency issues with hash map as far as I can see.

Reply:
One thread may create the local platformFontMap, then set the static SunFontManager.platformFontMap field, and then initialize the platformFontMap or mix this operations. The second thread may see non-null SunFontManager.platformFontMap which is not still initialized.

Reply:
Made `volatile` to guarantee memory visibility if thread saw non-null value.

Reply:
I do not any other questions, will file a bug and run the tests.